### DISCUSSION TITLE
Updating media.currentTime will reset it to 0 whatever the time is

### DISCUSSION THREAD

**Discussion by wfk007** [katspaugh/wavesurfer.js#3621] [Q&A] [answered]
@katspaugh After upgrading from 7.4.2 to 7.7.7, I encountered two problems:

1. `loadBlob` not working correctly: 
I finally found this is caused by this [PR](https://github.com/katspaugh/wavesurfer.js/pull/3471/files) with an empty mimeType in my project. I fixed it by giving it a valid mimeType, such as `audio/aac` and it worked!

2. Updating `media.currentTime` will reset it to 0 whatever the time is:

It's extraordinary and I haven't found any possible reasons either and can not reproduce it locally. Do you have any ideas for the updating from 7.4.2 to 7.7.7?
- 7.4.2 ✅
- 7.7.7 ❌

<img width="831" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/22289445/473fa934-c3f0-41d2-ab87-e3b664769828">

**Comment by katspaugh** [comment]
Sounds like issues related to the file format and/or your server handling this audio file. Current time setting involves [time range requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests) that the server must be able to handle.

Since it used to work in 7.4.2, pehaps we can make it work again for you. Please share the audio URL.

**Reply by wfk007** [reply]
The voice data is returned through the intranet interface and is information sensitive. Sorry, I cannot provide it.
The screenshot below is the response header of the interface:

<img width="544" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/22289445/74ef8fca-f238-4965-ad13-fb3c1b300d7f">

**Reply by katspaugh** [reply, marked as answer]
That doesn't help much. You could try `backend: 'WebAudio'` as a workaround.

**Reply by wfk007** [reply]
@katspaugh Thank you very much, it worked for me, and it would be even better if the voice is not distorted (robot voice) at high speeds.

**Comment by wfk007** [comment]
@katspaugh I finally found that the seek problem is caused by this [PR](https://github.com/katspaugh/wavesurfer.js/pull/3471/files) too. The difference is:

7.6.1 ✅
7.6.2 ❌

before: `media.src` is the blob src fetched by the wavesurfer
<img width="1056" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/22289445/f7563158-82ad-4b0e-83b7-eb0a73ce1f08">

after: `media.src` is a API path such as `/api/audio/download`
<img width="1100" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/22289445/51b850bc-2147-4859-b4a9-d8d45b762d7e">

I fix it by adding `Accept-Ranges: bytes` in the response header referred to this issue: https://github.com/katspaugh/wavesurfer.js/issues/492#issuecomment-745585545

**Reply by katspaugh** [reply]
Thanks for bisecting this!

I see. So when the src is set to a blob, it works even w/o an Accept-Ranges header, right? Which makes sense because it doesn't go to the server to fetch those bytes, they are all in the blob already.

So the problem boils down to serving the audio with the right content type. If you do that, you'll save bandwidth by not downloading the audio twice, and you won't need the Accept-Ranges header.

Especially with dynamic API URLs like `api/audio/download` without a file name and extension, your server should explicitely set the right content type for the browser to know it's a playable audio.
